### ISSUE TITLE
Cannot get custom request header in error handler.

### ISSUE THREAD

**Issue by ht-guo-yusen** [axios/axios#6149]
#### Summary

I add a header A  in request interceptor, and then when I got the 401 error, I want to get header A from the error.config.header, but I always got an undefined.
Even I print the log to console and confirmed that my header is there, but cannot get it.
![image](https://github.com/axios/axios/assets/146697471/1660a980-af92-4fd7-baab-3209e66f4b30)

![image](https://github.com/axios/axios/assets/146697471/7c1eec46-bb3a-4d37-9fc1-df1fad564f8f)

#### Environment
 - Axios Version [e.g. 0.18.0]
 - Adapter [e.g. XHR/HTTP]
 - Browser [e.g. Chrome, Safari]
 - Browser Version [e.g. 22]
 - Node.js Version [e.g. 13.0.1]
 - OS: [e.g. iOS 12.1.0, OSX 10.13.4]
 - Additional Library Versions [e.g. React 16.7, React Native 0.58.0]

**Comment by himittal09** [comment]
Can you confirm the issue by using the `.has()` and `.get()` method, that the header is really not accessible?

**Comment by ht-guo-yusen** [comment]
@himittal09 
yes, I tried again, .has() return false, and .get return undefined
![image](https://github.com/axios/axios/assets/146697471/4105ef3b-70f7-4877-b20c-184ccb70f738)

![image](https://github.com/axios/axios/assets/146697471/66532d1d-18d3-48e3-a56f-849c532749f7)

**Comment by jasonsaayman** [comment]
Thanks for your bug report, we will be closing this issue and ask that you please test and try re-create the issue with the latest version of Axios. If the bug is still present please open a issue. If possible and you are willing open a pull request with a proposed fix. Please see the motivation for doing this:

https://github.com/axios/axios/issues/6983
